clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/helm/HelmArtifactCredentials.java

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.helm;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Request.Builder;
+import com.squareup.okhttp.Response;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+@Data
+public class HelmArtifactCredentials implements ArtifactCredentials {
+  private final String name;
+  private final List<String> types = Arrays.asList("helm/chart");
+
+  @JsonIgnore
+  private final IndexParser indexParser;
+
+  @JsonIgnore
+  OkHttpClient okHttpClient;
+
+  @JsonIgnore
+  private final Builder requestBuilder;
+
+  @JsonIgnore
+  ObjectMapper objectMapper;
+
+  @Override
+  public boolean handlesType(String type) {
+    return type.equals("helm/chart");
+  }
+
+  public HelmArtifactCredentials(HelmArtifactAccount account, OkHttpClient okHttpClient) {
+    this.name = account.getName();
+    this.okHttpClient = okHttpClient;
+    this.indexParser = new IndexParser(account.getRepository());
+    Builder builder = new Builder();
+    boolean useLogin = !StringUtils.isEmpty(account.getUsername()) && !StringUtils.isEmpty(account.getPassword());
+    boolean useUsernamePasswordFile = !StringUtils.isEmpty(account.getUsernamePasswordFile());
+    boolean useAuth = useLogin || useUsernamePasswordFile;
+    if (useAuth) {
+      String authHeader = "";
+      if (useUsernamePasswordFile) {
+        authHeader = "Basic " + Base64.encodeBase64String((credentialsFromFile(account.getUsernamePasswordFile())).getBytes());
+      } else if (useLogin) {
+        authHeader = "Basic " + Base64.encodeBase64String((account.getUsername() + ":" + account.getPassword()).getBytes());
+      }
+      builder.header("Authorization", authHeader);
+      log.info("Loaded credentials for helm artifact account {}", account.getName());
+    } else {
+      log.info("No credentials included with helm artifact account {}", account.getName());
+    }
+    requestBuilder = builder;
+  }
+
+  private String credentialsFromFile(String filename) {
+    try {
+      String credentials = FileUtils.readFileToString(new File(filename));
+      return credentials.replace("\n", "");
```

Comment:
I would only strip leading/trailing whitespace or newlines just in case.

Reply:
@lwander credentialsFromFIle function has changed. Take a look again please. Thanks.